src/components/Form/Form.js

```diff
@@ -5,39 +5,51 @@ import { useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 
-export default function Form( { updateSalary, updateExpenses, updateZipcode} ) {
+export default function Form( { updateSalary, updateExpenses, updateZipcode, updateCombinedRate} ) {
 	const [salary, setSalary] = useState('');
   const [expenses, setExpenses] = useState('');
   const [zipcode, setZipcode] = useState('');
 	const navigate = useNavigate();
-
   function submitUserData(event) {
     event.preventDefault();
-		//do the setting state here
-		//after that's done, clearContent
-
-		updateSalary(salary);
-		updateExpenses(expenses);
-		updateZipcode(zipcode);
-
-		clearInput();
+		const validNumberRegex = /\d+/
+		const validZipcode = /^\d{5}(?:[-\s]\d{4})?$/gm;
```

Comment:
Nice regex!! I would strive to understand at a high level what each of these regular expressions are doing